go-controller/pkg/ovn/namespace.go

```diff
@@ -27,11 +28,14 @@ func (oc *Controller) syncNamespaces(namespaces []interface{}) {
 		expectedNs[ns.Name] = true
 	}
 
-	err := oc.forEachAddressSetUnhashedName(func(addrSetName,
-		namespaceName, nameSuffix string) {
+	err := oc.addressSetFactory.ForEachAddressSet(func(hashedAddrSetName, namespaceName, nameSuffix string) {
 		if nameSuffix == "" && !expectedNs[namespaceName] {
-			// delete the address sets for this namespace from OVN
-			deleteAddressSet(hashedAddressSet(addrSetName))
+			// delete the address sets for this policy from OVN
+			_, stderr, err := util.RunOVNNbctl("--if-exists", "destroy", "address_set", hashedAddrSetName)
```

Comment:
seems like this should be done via an `AddressSetFactory` method... Otherwise tests have to deal with setting up fake destroy commands even when using the fake address set factory

Reply:
@danwinship it used to be, but I removed it and re-added the nbctl calls to namespace/policy sync when I move AddressSet ownership to both namespaceInfo and gressPolicy. This was done so that ovnAddressSetFactory no longer had to keep a map of address sets, the only purpose of which was to track deletion.

I can certainly add it back, but I'd rather not add the set map back to ovnAddressSetFactory. I can docblock the DeleteAddressSet function to say you shouldn't call it on an OVN address set for which we have an AddressSet object.

Reply:
> I can docblock the DeleteAddressSet function to say you shouldn't call it on an OVN address set for which we have an AddressSet object.

But the restriction on calling `ovn-nbctl destroy address_set` on a "live" `AddressSet` exists whether or not `DeleteAddressSet` exists, right? At least this way we have somewhere to document it...

Reply:
let's not add the set back. sync function there's not much we can do. As @danwinship suggests you could just make a function on the factory that deletes address set by hash, so at least we could mock it out. Is that good enough? I mean the only reason the factory exists is so we can mock, so why not...

Also change it in the other place you call this in sync network policies to use the method.

Reply:
@trozet @danwinship I've added a delete function to the factory. Just worried that without a lot of docblock it'll get inadvertently used to remove an address set from OVN that is still known to ovnkube.